pkg/runtime/wfengine/component.go

```diff
@@ -80,8 +80,10 @@ func (c *workflowEngineComponent) Start(ctx context.Context, req *workflows.Star
 		return nil, errors.New("a workflow name is required")
 	}
 
+	// Init with capacity of 3 as worst-case scenario
+	opts := make([]api.NewOrchestrationOptions, 0, 3)
```

Comment:
What is the logic behind init with capacity 3?

Reply:
I added a comment that it's the "worst-case scenario". If you count the number of times something _could_ get appended to opts below, it's 3 at most. So this is just a micro-optimization

Reply:
Here we know it will have 3 or fewer elements, so it's better choice to pre-allocate the necessary memory and Go doesn't need to reallocate memory every time we append an element to the slice.